### ISSUE TITLE
[💡SUG] Why set scores of history items to -inf as the target item may be included in the historical interactions

### ISSUE THREAD

**Issue by zuoxingdong** [RUCAIBox/RecBole#1823]
The context of this question is the evaluation of sequential recommendation models. 
Currently [the scores of history items are set to be `-inf`](https://github.com/RUCAIBox/RecBole/blob/a757b7c31bfa407c82847b4a06c8580fa1a6b45e/recbole/utils/case_study.py#L73), however, what if the target item has already be included in the historical interaction sequence (i.e. taken as inputs to the model)?

**Comment by Sherry-XLL** [comment]
Hello @zuoxingdong, thanks for your attention to RecBole!

For the [evaluation](https://recbole.io/docs/user_guide/config/evaluation_settings.html) of recommendation models, we have set the parameter `repeatable (bool)` to decide whether to evaluate the result with a repeatable recommendation scene. Note that it is disabled for sequential models as the recommendation is already repeatable. For other models, defaults to `False`.

In other words, for sequential recommendation models, we do not set historical items to be `-inf`, so that during the testing phase, we can still recommend items that have interacted in the past, which can be repeated. For other models, historical items will be excluded from the recommendation.

Feel free to contact us if you have further questions.
